### PR TITLE
 repository pattern and add student with course selected

### PR DESCRIPTION
**Pull request by hamzaaslam76**

### CODE REVIEW

File: StudentApplication/Repository/Repository/StudentRepositoryP.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                try
+                {
+                    var students = (from s in DbContext.students
+                                    join sc in DbContext.studentCourses on s.StudentId equals sc.StudentId
+                                    group s by s.Studentname into grouped
+                                    select (new StudentDto())
+                                   
+                                    ).ToList();
+                    
+
```
Comment (Shanfiaz) [inline StudentApplication/Repository/Repository/StudentRepositoryP.cs:63]: Remove the extra spacings

File: StudentApplication/Repository/Repository/StudentRepositoryP.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                                   
+                                    ).ToList();
+                    
+                    
+                   
+
+                }
+                catch (Exception e)
+                {
+
```
Comment (Shanfiaz) [inline StudentApplication/Repository/Repository/StudentRepositoryP.cs:69]: Why empty?????

File: StudentApplication/Repository/Repository/_IAllRepository.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+namespace Repository.Repository
+{
+    public interface _IAllRepository<T> where T:class
+    {
+        IEnumerable<T> GetModel();
+        T GetModelById(int Modelid);
+        bool InsertModel(T Model);
+        bool DeleteModel(int Model);
+        bool UpdateModel(T Model);
+        void    Save();
```
Comment (Shanfiaz) [inline StudentApplication/Repository/Repository/_IAllRepository.cs:16]: remove the spac between void and save

File: StudentApplication/WebApi/Controllers/StudentsController.cs [hunk 4]
```diff
@@ -12,58 +12,98 @@ namespace WebApi.Controllers
...
         {
-            var item = _studentRepository.AddStudent(addstd);
-            if(item == true)
+           
+            var item = _studentRepositoryp.AddStudent(studentes,courseID);
+            if (item == true)
             {
                 return Ok(item);
             }
+
```
Comment (Shanfiaz) [inline StudentApplication/WebApi/Controllers/StudentsController.cs:31]: remove spacings

File: StudentApplication/WebApi/Controllers/StudentsController.cs [hunk 5]
```diff
@@ -12,58 +12,98 @@ namespace WebApi.Controllers
...
+                var check = _studentRepositoryp.DeleteModel(id);
+                return Ok(check);
             }
-            return BadRequest("Student not Deleted");
+            catch(Exception e)
+            {
+                return Ok("Student not Deleted");
+            }
+              
+
```
Comment (Shanfiaz) [inline StudentApplication/WebApi/Controllers/StudentsController.cs:69]: Remove spaces

File: StudentApplication/WebApi/Controllers/StudentsController.cs [hunk 6]
```diff
@@ -12,58 +12,98 @@ namespace WebApi.Controllers
...
+            //}
             return BadRequest("Data Not Found");
         }
 
         [HttpGet]
         
         public IHttpActionResult getStudentById(int id)
         {
-            var Item = _studentRepository.SearchStudent(id);
+            var Item = _studentRepositoryp.GetModelById(id);
```
Comment (Shanfiaz) [inline StudentApplication/WebApi/Controllers/StudentsController.cs:106]: var Item name is in pascal case, it should be camel case

File: script/WebApiShowTable.js [hunk 7]
```diff
@@ -1,13 +1,18 @@
-var Url="https://localhost:44319/api/Students/";
+//var BaseUrl="https://localhost:44319/api/";
+
```
Comment (Shanfiaz) [inline script/WebApiShowTable.js:2]: remove extra spacings

File: script/WebApiShowTable.js [hunk 8]
```diff
@@ -1,13 +1,18 @@
...
+
+var BaseUrl="https://localhost:44319/api/";
 
 //Student List Show Method
 function GetRecordFromWebApi()
 {
+    debugger;
```
Comment (Shanfiaz) [inline script/WebApiShowTable.js:9]: remove debuggers

File: StudentApplication/Repository/Repository/StudentRepositoryP.cs (hunk 1 above)
Comment (hamzaaslam76) [reply StudentApplication/Repository/Repository/StudentRepositoryP.cs:63]: OK

File: StudentApplication/Repository/Repository/StudentRepositoryP.cs [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+                {
+                   
+                    InsertModel(std);
+                    StudentCourse stdCourse = new StudentCourse();
+                    stdCourse.StudentId = std.StudentId;
+                    foreach (var course in courseList)
+                    {
+                        stdCourse.CourseId =Convert.ToInt32(course);
+                        DbContext.studentCourses.Add(stdCourse);
+                        DbContext.SaveChanges();
```
Comment (Nauman7) [inline StudentApplication/Repository/Repository/StudentRepositoryP.cs:33]: move this savechanges outside the foreach loop

File: StudentApplication/Repository/Repository/StudentRepositoryP.cs [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+          
+            using (var DbContext = new context())
+            {
+                try
+                {
+                    var students = (from s in DbContext.students
+                                    join sc in DbContext.studentCourses on s.StudentId equals sc.StudentId
+                                    group s by s.Studentname into grouped
+                                    select (new StudentDto())
+
```
Comment (Nauman7) [inline StudentApplication/Repository/Repository/StudentRepositoryP.cs:60]: why not using the RepositoryBase Method? it seems empty model in select

File: StudentApplication/WebApi/Controllers/StudentsController.cs [hunk 11]
```diff
@@ -12,58 +12,98 @@ namespace WebApi.Controllers
...
 
         [HttpGet]
 
         public IHttpActionResult getStudent()
         {
-            var item = _studentRepository.getStudentList();
-            return Ok(item);
+            try
+            {
+                var item = _studentRepositoryp.GetModel();
```
Comment (Nauman7) [inline StudentApplication/WebApi/Controllers/StudentsController.cs:42]: make it in one line
`return itme==null?Ok("No Record In Table"):Ok(item) `

File: StudentApplication/WebApi/Controllers/StudentsController.cs [hunk 12]
```diff
@@ -12,58 +12,98 @@ namespace WebApi.Controllers
...
+                    return Ok(_studentRepositoryp.UpdateModel(student));
+                }
             }
+            catch(Exception e)
+            {
+
+            }
+            
+
+            //if(item == true)
```
Comment (Nauman7) [inline StudentApplication/WebApi/Controllers/StudentsController.cs:95]: remove commented code

File: WebApiStudent.html [hunk 13]
```diff
@@ -1,4 +1,147 @@
...
+    $('#myselect').select2({
+        placeholder:"choose Courses"
+    });
+GetCourseInTable();
+</script>
+
+</body>
+</html>
+
+
```
Comment (Nauman7) [inline WebApiStudent.html:96]: why empty lines ???

File: script/WebApiShowTable.js [hunk 14]
```diff
@@ -173,66 +162,101 @@ function getrecordaddsave() {
...
-    });
-    
-    
+            var Url = BaseUrl + "Students/"; 
+            var ar = selectco.toString();
+            AjaxMethodFunction(Url+"?courseID="+ ar , student , 'POST', onSuccess);
+            function onSuccess(data)
+            {
+               
+               if(data == true)
```
Comment (Nauman7) [inline script/WebApiShowTable.js:179]: if(data) is enough

**Review by Nauman7 — CHANGES_REQUESTED** [review]

**Pull request merged**